core/src/main/scala/kafka/api/ApiVersion.scala

```diff
@@ -256,5 +258,12 @@ case object KAFKA_2_1_IV0 extends DefaultApiVersion {
   val shortVersion: String = "2.1"
   val subVersion = "IV0"
   val recordVersion = RecordVersion.V2
+  val id: Int = 17
```

Comment:
I guess it is probably not a problem for most users and it is cleaner to make this change.

One question is, if user is using the trunk version where KAFKA_2_1_IV0 has id=18, and then they bump up to the trunk version where KAFKA_2_1_IV0 has id=17 and KAFKA_2_1_IV1 has id=18, would it cause problem?

Reply:
As far as I can tell, we only use the id for ordering internally, so I don't think this change has any impact.